proxy/pkg/zdmproxy/cqlconn.go

```diff
@@ -237,6 +239,8 @@ func (c *cqlConn) InitializeContext(version primitive.ProtocolVersion, ctx conte
 		return fmt.Errorf("failed to perform handshake: %w", err)
 	}
 
+	c.protocolVersion = &atomic.Value{}
```

Comment:
I think we should set this field in `NewCqlConnection` and set the initial protocol version there as well so there's no confusion as to whether this field is being modified after the connection object is created.